app/models/text.rb

```diff
@@ -27,7 +27,7 @@ def slug
 
   # Used for serialisation
   def attributes
-    { title: nil, ordinal: nil, description: nil, body: nil, authors: [], free: false, episode_type: nil }.stringify_keys
+    { title: nil, ordinal: nil, description: nil, body: nil, authors: [], free: false, episode_type: nil, ref: nil }.stringify_keys
```

Comment:
Should this here be something else than `episode_type`? 
You've changed `episode` to `segment` above?

Reply:
We can. I retained the existing naming of episode here for now rather than creating a new segment_type as it uses existing parser of video, chapter and assessment.

I think i should include it, just to be safe.